src/course-unit/course-xblock/CourseXBlock.jsx

```diff
@@ -5,24 +5,30 @@ import {
 } from '@edx/paragon';
 import { EditOutline as EditIcon, MoreVert as MoveVertIcon } from '@edx/paragon/icons';
 import { useIntl } from '@edx/frontend-platform/i18n';
-import { useDispatch } from 'react-redux';
+import { useDispatch, useSelector } from 'react-redux';
+import { useNavigate } from 'react-router-dom';
 
 import DeleteModal from '../../generic/delete-modal/DeleteModal';
 import ConfigureModal from '../../generic/configure-modal/ConfigureModal';
 import { scrollToElement } from '../../course-outline/utils';
 import { COURSE_BLOCK_NAMES } from '../../constants';
 import { copyToClipboard } from '../data/thunk';
+import { COMPONENT_ICON_TYPES } from '../constants';
 import ContentIFrame from './ContentIFrame';
 import { getIFrameUrl } from './urls';
 import messages from './messages';
 
+import { getCourseId } from '../data/selectors';
+
 const CourseXBlock = ({
-  id, title, unitXBlockActions, shouldScroll, userPartitionInfo, handleConfigureSubmit, ...props
+  id, title, type, unitXBlockActions, shouldScroll, userPartitionInfo, handleConfigureSubmit, ...props
 }) => {
   const courseXBlockElementRef = useRef(null);
   const [isDeleteModalOpen, openDeleteModal, closeDeleteModal] = useToggle(false);
   const [isConfigureModalOpen, openConfigureModal, closeConfigureModal] = useToggle(false);
   const dispatch = useDispatch();
+  const navigate = useNavigate();
+  const courseId = useSelector(getCourseId);
```

Comment:
[proposal]: Let's pass the `courseId` through props. I think this is a good idea because we don't have deep nesting and can use the `courseId` passed from the component above

Reply:
I think here it doesn't make any effect because adding it to props will mean that we would want to reuse it later in different place. Right now we won't consider this.
Global state is more suitable here because we can avoid some props drilling.